Source/CesiumRuntime/Public/GlobeAwareDefaultPawn.h

```diff
@@ -220,7 +220,9 @@ class CESIUMRUNTIME_API AGlobeAwareDefaultPawn : public ADefaultPawn {
 
   // the current ECEF coordinates, stored in case they need to be restored on
   // georeference update
-  glm::dvec3 _currentEcef;
+  UPROPERTY()
```

Comment:
Is there a specific reason to make this a `UPROPERTY`? 

From a first test, this does not seem to be necessary: Letting this be just a `glm::dvec3` seemed to be sufficient, and it appears as if only the `this->_currentEcef = this->GetECEFCameraLocation();` call had been missing

Reply:
Nah, all this depends on whether "Auto-Possess" is enabled. Without Auto-Possess, it creates **two** pawn instances. One of them is in the world, and the other one not, so one of them has a Georeference, and the other one not. Yikes.

Reply:
@javagl Can you elaborate on how auto-possess creates a different pawn, without a georeference, and not in the world? I just tried to remove the uproperty as you suggested, and dropped in a new dynamicPawn (in case the change corrupted the old one), and it still disappeared into space (or maybe the center of earth hahah). The problem does not seem to exist when it's a UPROPERTY as it now is, although I'm not positive I understand completely why that's the case.

Reply:
Yeah, it's difficult: When hitting "Play" **with** "Auto-possess", then there is one pawn, and it is at the right location (using the first map from the samples here) : 

![CesiumUnreal Pawn With Auto-possess](https://user-images.githubusercontent.com/5597569/126800755-8ab1bd43-5e60-438a-8ed1-f882f02dc525.png)

But **without** "Auto-possess", there are two pawns, and they are both at odd locations:

![CesiumUnreal Pawn WithOUT Auto-possess](https://user-images.githubusercontent.com/5597569/126801038-ee40afb6-31f1-4ac5-a460-392c28926a30.png)

**Very** roughly: When hitting "Play", it will internally "clone" some objects and put them into the "Play-World", and add other object (the ones shown in yellow in the world outliner). _Without_ "Auto-possess", it just adds the DynamicPawn as it is (the white one), and creates a new one (the yellow one) that can actually be controlled. This can be observed when selecting them and moving around: The position of the yellow one is immediately updated to reflect the camera position. 

The position of the white one only changes during Origin Rebasing. And I was a bit surprised to see that Origin Rebasing is not handled explicitly in `GlobeAwareDefaultPawn`. I thought that it might need an overridden `ApplyWorldOffset` method, to update the `_currentEcef` or so, but maybe that's not how this is supposed to work...

Unfortunately, there is another "dimension" of that problem: If I remember correctly, then there is this "Game Mode" that contains something like a "Default Pawn Class to Use", and it _might_ be that, depending on that setting, it might instantiate different types of pawns (as the "yellow" ones) when starting to play without auto-possess.

All this are reasons why I thought that it could make sense to have a closer look at the `GlobeAwareDefaultPawn` (besides smaller things, like the TODO in `BeginPlay`, and some overridden methods that only call or re-implement the `super` version...)